build: Bump golangci-lint from 1.31.0 to 1.41.1

Mini256 opened the pull request:
The old version hasn't ARM architecture binary package for Mac (M1).

hi-rustin reviewed (changes requested):
Do we need update the GitHub actions?

Mini256 commented:
Some plugins have been discarded in the new version.

```
WARN [runner] The linter 'scopelint' is deprecated (since v1.39.0) due to: The repository of the linter has been deprecated by the owner.  Replaced by exportloopref. 
WARN [runner] The linter 'golint' is deprecated (since v1.41.0) due to: The repository of the linter has been archived by the owner.  Replaced by revive. 
WARN [runner] The linter 'interfacer' is deprecated (since v1.38.0) due to: The repository of the linter has been archived by the owner.
```


```
cmd/check-external-plugin-config/main.go:75:2: if-return: redundant if ...; err != nil check, just return error instead. (revive)
        if err := config.Validate(); err != nil {
                return err
        }
internal/pkg/externalplugins/config.go:537:2: if-return: redundant if ...; err != nil check, just return error instead. (revive)
        if err := validateTars(c.TiCommunityTars); err != nil {
                return err
        }
```

Mini256 commented:
> Do we need update the GitHub actions?

Updated it.

hi-rustin reviewed (approved):
Cool!

hi-rustin commented:
/merge
